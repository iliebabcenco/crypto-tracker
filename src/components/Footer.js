import React from 'react';
import '../styles/Footer.css';

const Footer = () => (
  <footer className="footer">
    Copyright
    {' '}
    &copy;
    {' '}
    <a href="https://github.com/iliebabcenco" alt="github link">Ilie Babcenco</a>
  </footer>
);

export default Footer;
