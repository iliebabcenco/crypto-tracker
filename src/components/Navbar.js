import SearchForm from '../containers/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
