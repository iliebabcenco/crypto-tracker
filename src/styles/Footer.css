.footer {
  color: white;
  margin: 0 auto;
  padding-top: 2rem;
  padding-bottom: 2rem;
  border-top: 1px solid #c9ab81;
  background-color: rgba(21, 30, 36, 0.8);
  text-align: center;
  width: 100%;
  position: absolute;
  left: 0;
  bottom: 0;
  right: 0;
}