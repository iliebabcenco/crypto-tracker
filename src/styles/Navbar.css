.navbar {
  width: 100%;
  color: white;
  display: flex;
  margin: 0 auto;
  padding-bottom: 1rem;
  padding-top: 1rem;
  justify-content: space-around;
  border-bottom: 1px solid #c9ab81;
  background-color: rgba(21, 30, 36, 0.8);
}