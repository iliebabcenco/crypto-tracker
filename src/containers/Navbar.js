import SearchForm from '../components/SearchForm';
import '../styles/Navbar.css';

const Navbar = () => (
  <nav className="navbar">
    <h1 className="nav-title">CryptoTracker</h1>
    <SearchForm />
  </nav>
);

export default Navbar;
