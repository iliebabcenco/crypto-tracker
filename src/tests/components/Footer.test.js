import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Footer from '../../components/Footer';
import store from '../../store/store';
import Navbar from '../../containers/Navbar';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Provider store={store}><Footer /></Provider>);
  expect(asFragment(<Provider store={store}><Footer /></Provider>)).toMatchSnapshot();
});

it('displays correct first child', () => {
  const { container } = render(<Provider store={store}><Footer /></Provider>);
  expect(container.firstChild).toMatchSnapshot();
});

it('negative check', () => {
  render(<Provider store={store}><Navbar /></Provider>);
  expect(screen.queryByText('Letâ€™s do some math!')).not.toBeInTheDocument();
});
