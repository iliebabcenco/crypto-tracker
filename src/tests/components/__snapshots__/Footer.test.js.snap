// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correct first child 1`] = `
<footer
  class="footer"
>
  Copyright
   
  ©
   
  <a
    alt="github link"
    href="https://github.com/iliebabcenco"
  >
    Ilie Babcenco
  </a>
</footer>
`;

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    Copyright © 
    <a
      alt="github link"
      href="https://github.com/iliebabcenco"
    >
      Ilie Babcenco
    </a>
  </footer>
</DocumentFragment>
`;
